aws_lambda_builders/workflows/go_dep/workflow.py

```diff
@@ -0,0 +1,57 @@
+"""
+Go Dep Workflow
+"""
+
+from aws_lambda_builders.workflow import BaseWorkflow, Capability
+from aws_lambda_builders.actions import CopySourceAction, FileCleanupAction
+from .utils import OSUtils
+from .exec import SubprocessExec
+from .actions import GoDepInstallAction, GoBuildAction
+
+class GoDepWorkflow(BaseWorkflow):
+
+    """
+    A Lambda builder workflow that knows how to
+    package Go projects using Dep
+    """
+
+    NAME = "GoDepBuilder"
+
+    CAPABILITY = Capability(language="go",
+                            dependency_manager="dep",
+                            application_framework=None)
+
+    EXCLUDED_FILES = (".aws-sam")
+
+    def __init__(self,
+                 source_dir,
+                 artifacts_dir,
+                 scratch_dir,
+                 manifest_path,
+                 runtime=None,
+                 osutils=None,
+                 **kwargs):
+        super(GoDepWorkflow, self).__init__(source_dir,
+                                                artifacts_dir,
+                                                scratch_dir,
+                                                manifest_path,
+                                                runtime=runtime,
+                                                **kwargs)
+
+        if osutils is None:
+            osutils = OSUtils()
+
+        subprocess_dep = SubprocessExec(osutils, "dep")
+        subprocess_go = SubprocessExec(osutils, "go")
+
+        tar_dest_dir = osutils.joinpath(scratch_dir, 'unpacked')
+        tar_package_dir = osutils.joinpath(tar_dest_dir, 'package')
+
+        handler = kwargs.get("options", {}).get("handler", None)
+
+        self.actions = [
+            CopySourceAction(tar_package_dir, artifacts_dir, excludes=self.EXCLUDED_FILES),
+            GoDepInstallAction(artifacts_dir, subprocess_dep),
+            GoBuildAction(artifacts_dir, handler, subprocess_go),
+            FileCleanupAction(artifacts_dir, r'/([\w]+\.go|Gopkg.toml)/gi')
```

Comment:
Why aren't we doing actions in the scratch directory and then doing a copy of the binary produced to the file resting place?

Doing regex's to clean up files isn't a process I think we should follow at all here. This is the reason we have a scratch directory.